### ISSUE TITLE
forge scripts: "Failed to decode constructor arguments"

### ISSUE THREAD

**Issue by clabby** [foundry-rs/foundry#6571]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (0ae39ea 2023-12-11T00:27:32.487222000Z)

### What command(s) is the bug in?

forge script

### Operating System

macOS (Apple Silicon)

### Describe the bug

On the last few nightly releases of foundry, I've been receiving the error:

```
2023-12-11T18:49:05.565916Z ERROR forge::cmd::script::transaction: Failed to decode constructor arguments constructor="constructor(address,string)" contract=Some("ResolvedDelegateProxy") bytecode="0x..."
```

This seems to be a regression of the fix introduced for https://github.com/foundry-rs/foundry/issues/3229, and I can confirm that it was working on [last week's nightly release](https://github.com/foundry-rs/foundry/releases/tag/nightly-5b7e4cb3c882b28f3c32ba580de27ce7381f415a).

### Steps to Reproduce
**Before Regression**
```
foundryup --version nightly-5b7e4cb3c882b28f3c32ba580de27ce7381f415a && \
  git clone git@github.com:ethereum-optimism/optimism.git && \
  cd optimism && \
  make devnet-allocs
```

**After Regression**
```
foundryup && \
  git clone git@github.com:ethereum-optimism/optimism.git && \
  cd optimism && \
  make devnet-allocs
```

**Comment by mds1** [comment]
From bisecting the above steps to repro it seems the issue was introduced in https://github.com/foundry-rs/foundry/pull/6300, cc @klkvr

**Comment by mds1** [comment]
Closed by #6589, thanks again @klkvr!
